easybuild/framework/easyblock.py

```diff
@@ -1728,19 +1743,35 @@ def run_step(self, step, methods, skippable=False):
         """
         Run step, returns false when execution should be stopped
         """
-        if skippable and (self.skip or step in self.cfg['skipsteps']):
-            self.log.info("Skipping %s step" % step)
+        only_module = build_option('only_module')
+        force = build_option('force')
+        skip = False
+
+        # skip step if specified, either as individual (skippable) step, or when only generating module file
+        # still run sanity check when only generating module
+        skip_individual_step = skippable and (self.skip or step in self.cfg['skipsteps'])
+        only_module_skip = only_module and not step in ['sanitycheck', 'module']
+        if skip_individual_step or only_module_skip:
+            self.log.info("Skipping %s step", step)
+            skip = True
+
+        # allow skipping sanity check too when only generating module and --force is enable
+        elif only_module and step == 'sanitycheck' and force:
+            self.log.info("Skipping %s step, due to combo of --only-module and --force", step)
```

Comment:
don't use commandline options in the message, they are only maintained in the `options.py`